tunnel.ForwardPortE no longer works with named targetPort

**Describe the bug**

The recent fix to `tunnel.ForwardPortE` in https://github.com/gruntwork-io/terratest/pull/1547 broke port selection by name.

**To Reproduce**

Edit `modules/k8s/tunnel_test.go` and edit `ExamplePodWithServiceYAMLTemplate`. Add a name to the containerPort and change targetPort in the service to that name.

```diff
diff --git a/modules/k8s/tunnel_test.go b/modules/k8s/tunnel_test.go
index 79a3ce6a..7a9bc534 100644
--- a/modules/k8s/tunnel_test.go
+++ b/modules/k8s/tunnel_test.go
@@ -134,6 +134,7 @@ spec:
     image: nginx:1.15.7
     ports:
     - containerPort: 80
+      name: http
     readinessProbe:
       httpGet:
         path: /
@@ -149,6 +150,6 @@ spec:
     app: nginx
   ports:
   - protocol: TCP
-    targetPort: 80
+    targetPort: http
     port: 8080
```

```bash
go test -run TestTunnelOpensAPortForwardTunnelToService --tags kubernetes

--- FAIL: TestTunnelOpensAPortForwardTunnelToService (7.39s)
    tunnel.go:173: 
                Error Trace:    /home/sdiez/projects/github.com/gruntwork-io/terratest/modules/k8s/tunnel.go:173
                                                        /home/sdiez/projects/github.com/gruntwork-io/terratest/modules/k8s/tunnel_test.go:95
                Error:          Received unexpected error:
                                remote port must be > 0
                Test:           TestTunnelOpensAPortForwardTunnelToService
FAIL
exit status 1
FAIL    github.com/gruntwork-io/terratest/modules/k8s   7.401s
```

**Expected behavior**

Port selection through service works with named and numbered targetPort


**Nice to have**
- [x] Terminal output
- [ ] Screenshots

**Versions**
- Terratest version:  v0.49.0
- Environment details (Ubuntu 20.04, Windows 10, etc.): Fedora 41, minikube version: v1.35.0

**Additional context**
I will add a PR to handle this soon.

--- a/modules/k8s/tunnel.go
+++ b/modules/k8s/tunnel.go
@@ -16,7 +16,9 @@ import (
 	"sync"
 
 	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
 	"k8s.io/client-go/tools/portforward"
 	"k8s.io/client-go/transport/spdy"
 
@@ -220,6 +222,19 @@ func (tunnel *Tunnel) ForwardPortE(t testing.TestingT) error {
 		var portFound = false
 		for _, portSpec := range service.Spec.Ports {
 			if portSpec.Port == int32(targetPort) {
+				if portSpec.TargetPort.Type == intstr.String {
+					pod, err := GetPodE(t, tunnel.kubectlOptions, podName)
+					if err != nil {
+						return err
+					}
+					targetPort, err = getPodPortByName(pod, portSpec.TargetPort.String())
+					if err != nil {
+						tunnel.logger.Logf(t, "Error selecting port by name: %s", err)
+						return err
+					}
+					portFound = true
+					break
+				}
 				targetPort = portSpec.TargetPort.IntValue()
 				portFound = true
 				break
@@ -322,4 +337,18 @@ func GetAvailablePortE(t testing.TestingT) (int, error) {
 		return 0, err
 	}
 	return port, err
+}
+
+func getPodPortByName(pod *corev1.Pod, portName string) (int, error) {
+	if pod == nil {
+		return 0, errors.New("cannot get port for pod which is nil")
+	}
+	for _, container := range pod.Spec.Containers {
+		for _, port := range container.Ports {
+			if port.Name == portName {
+				return int(port.ContainerPort), nil
+			}
+		}
+	}
+	return 0, fmt.Errorf("could not find port %s in pod %s", portName, pod.Name)
 }

--- a/modules/k8s/tunnel_test.go
+++ b/modules/k8s/tunnel_test.go
@@ -83,29 +83,55 @@ func TestTunnelOpensAPortForwardTunnelToService(t *testing.T) {
 
 	uniqueID := strings.ToLower(random.UniqueId())
 	options := NewKubectlOptions("", "", uniqueID)
-	configData := fmt.Sprintf(ExamplePodWithServiceYAMLTemplate, uniqueID, uniqueID, uniqueID)
-	defer KubectlDeleteFromString(t, options, configData)
+	configData := fmt.Sprintf(ExamplePodWithServiceYAMLTemplate, uniqueID, uniqueID, uniqueID, uniqueID)
+	t.Cleanup(func() {
+		KubectlDeleteFromString(t, options, configData)
+	})
 	KubectlApplyFromString(t, options, configData)
+	// t.FailNow()
 	WaitUntilPodAvailable(t, options, "nginx-pod", 60, 1*time.Second)
-	WaitUntilServiceAvailable(t, options, "nginx-service", 60, 1*time.Second)
 
-	// Open a tunnel from any available port locally
-	tunnel := NewTunnel(options, ResourceTypeService, "nginx-service", 0, 8080)
-	defer tunnel.Close()
-	tunnel.ForwardPort(t)
+	testCases := []struct {
+		name        string
+		serviceName string
+	}{
+		{
+			"Pod target port by number",
+			"nginx-service-number",
+		},
+		{
+			"Pod target port by name",
+			"nginx-service-name",
+		},
+	}
 
-	// Setup a TLS configuration to submit with the helper, a blank struct is acceptable
-	tlsConfig := tls.Config{}
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			t.Parallel()
 
-	// Try to access the nginx service on the local port, retrying until we get a good response for up to 5 minutes
-	http_helper.HttpGetWithRetryWithCustomValidation(
-		t,
-		fmt.Sprintf("http://%s", tunnel.Endpoint()),
-		&tlsConfig,
-		60,
-		5*time.Second,
-		verifyNginxWelcomePage,
-	)
+			WaitUntilServiceAvailable(t, options, testCase.serviceName, 60, 1*time.Second)
+
+			// Open a tunnel from any available port locally
+			tunnel := NewTunnel(options, ResourceTypeService, testCase.serviceName, 0, 8080)
+			t.Cleanup(func() {
+				tunnel.Close()
+			})
+			tunnel.ForwardPort(t)
+
+			// Setup a TLS configuration to submit with the helper, a blank struct is acceptable
+			tlsConfig := tls.Config{}
+
+			// Try to access the nginx service on the local port, retrying until we get a good response for up to 5 minutes
+			http_helper.HttpGetWithRetryWithCustomValidation(
+				t,
+				fmt.Sprintf("http://%s", tunnel.Endpoint()),
+				&tlsConfig,
+				60,
+				5*time.Second,
+				verifyNginxWelcomePage,
+			)
+		})
+	}
 }
 
 func verifyNginxWelcomePage(statusCode int, body string) bool {
@@ -134,6 +160,7 @@ spec:
     image: nginx:1.15.7
     ports:
     - containerPort: 80
+      name: http
     readinessProbe:
       httpGet:
         path: /
@@ -142,7 +169,7 @@ spec:
 apiVersion: v1
 kind: Service
 metadata:
-  name: nginx-service
+  name: nginx-service-number
   namespace: %s
 spec:
   selector:
@@ -150,5 +177,18 @@ spec:
   ports:
   - protocol: TCP
     targetPort: 80
+    port: 8080
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: nginx-service-name
+  namespace: %s
+spec:
+  selector:
+    app: nginx
+  ports:
+  - protocol: TCP
+    targetPort: http
     port: 8080
 `

